components/sup/src/main.rs

```diff
@@ -314,6 +324,73 @@ fn sub_start(m: &ArgMatches) -> Result<()> {
     Ok(())
 }
 
+fn sub_status(m: &ArgMatches) -> Result<()> {
+    if m.is_present("VERBOSE") {
+        sup::output::set_verbose(true);
+    }
+    if m.is_present("NO_COLOR") {
+        sup::output::set_no_color(true);
+    }
+    let cfg = mgrcfg_from_matches(m)?;
+    let mut services: Vec<ServiceStatus> = Vec::new();
+    let mut query_one = false;
+
+    if Manager::is_running(&cfg)? {
+        match m.value_of("PKG_IDENT") {
+            Some(pkg) => {
+                query_one = true;
+                match Manager::service_status(cfg, PackageIdent::from_str(pkg)?) {
+                    Ok(status) => {
+                        services.push(status);
+                    }
+                    Err(_) => {
+                        outputln!("The {} service is not currently loaded.", pkg);
+                        std::process::exit(5);
+                    }
+                }
+            }
+            None => services.append(&mut Manager::status(cfg)?),
+        }
+
+        for status in services {
+            match status.supervisor.state {
+                ProcessState::Up => {
+                    outputln!("The {} service entered the running state with PID: {} {} seconds ago.",
+                              status.package,
+                              status.supervisor.pid.unwrap(),
+                              status.supervisor.elapsed.num_seconds())
+                }
+                ProcessState::Down => {
+                    outputln!("The {} service is not currently running.", status.package);
+                    if query_one {
+                        std::process::exit(2);
+                    }
+                }
+                ProcessState::Start => {
+                    outputln!("The {} service entered the  starting state {} seconds ago.",
+                              status.package,
+                              status.supervisor.elapsed.num_seconds());
+                    if query_one {
+                        std::process::exit(3);
+                    }
+                }
+                ProcessState::Restart => {
+                    outputln!("The {} service entered the restarting state {} seconds ago.",
+                              status.package,
+                              status.supervisor.elapsed.num_seconds());
+                    if query_one {
+                        std::process::exit(4);
+                    }
+                }
+            }
+        }
+    } else {
```

Comment:
Instead of an `if`/`else` block I'd recommend using a _function short-circuit_ where you return out of the function early.

```rust
if !Manager::is_running(&cfg)? {
    std::process::exit(6);
}

// ... rest of code
```

This would be a design pattern that we have in our coding standards had anyone ever gotten around to writing them yet ;)